R/csv.R

```diff
@@ -281,8 +283,13 @@ read_cmdstan_csv <- function(files,
       draws_list_id <- length(draws) + 1
       warmup_draws_list_id <- length(warmup_draws) + 1
       if (metadata$method == "pathfinder") {
-        metadata$variables = union(metadata$sampler_diagnostics, metadata$variables)
-        variables = union(metadata$sampler_diagnostics, variables)
+        metadata$variables <- union(metadata$sampler_diagnostics, metadata$variables)
+        if (!user_variables_subset) {
+          # because for pathfinder variables and diagnostics are read in together,
+          # if user hasn't selected a custom subset of variables we need to include
+          # all diagnostics
+          variables <- union(metadata$sampler_diagnostics, variables)
+        }
```

Comment:
Pathfinder is a special case because diagnostics and variables are read in together (mcmc reads them in separately). So some special handling was needed to subset properly for pathfinder.